go-controller/pkg/ovn/secondary_layer2_network_controller.go

```diff
@@ -456,7 +458,14 @@ func (oc *SecondaryLayer2NetworkController) addUpdateLocalNodeEvent(node *corev1
 					errs = append(errs, err)
 					oc.gatewaysFailed.Store(node.Name, true)
 				} else {
-					oc.gatewaysFailed.Delete(node.Name)
+					if util.IsNetworkSegmentationSupportEnabled() && oc.IsPrimaryNetwork() {
```

Comment:
is this needed ? Shouldn't this actually be part of the condition for `nSyncs.syncGw` ?

Reply:
oh I remember why.. :)
because both L3 and localnet don't need it in the gateway.... its an L2 only thing hence had to put it here..
it was based on your previous comment in my previous PR